core/src/main/scala/scalacache/package.scala

```diff
@@ -27,6 +27,10 @@ package object scalacache extends StrictLogging {
       _caching(keyParts: _*)(Some(ttl))(f)
     }
 
+    def cachingWithOptionalTTL(keyParts: Any*)(optionalttl: Option[Duration])(f: => Future[V])(implicit flags: Flags, execContext: ExecutionContext = ExecutionContext.global): Future[V] = {
```

Comment:
For consistency with other methods, please add a wrapper for this in the top-level of the package object, similar to `cachingWithTTL` on line 225. (Sorry, it's a little confusing - there's a typed API and an untyped wrapper, with sync and async variants, meaning a total of 4 places to add new methods!)